saleor/checkout/utils.py

```diff
@@ -81,6 +114,61 @@ def invalidate_checkout_prices(
     return updated_fields
 
 
+def apply_gift_reward_if_applicable(checkout: "Checkout") -> None:
+    """Apply gift reward if applicable on newly created checkout.
+
+    This method apply the gift reward if any gift promotion exists and
+    when it's giving the best discount on the current checkout.
+    """
+    if not PromotionRule.objects.filter(gifts__isnull=False).exists():
```

Comment:
Maybe we could also add a channel param here? Could we skip some calculations if there is no PromotnionRule with gifts in the channel instead of in the shop?